lib/manageiq_performance/configuration.rb

```diff
@@ -143,4 +143,16 @@ def browser_mode_config(opts={})
   def self.config
     @config ||= Configuration.load_config
   end
+
+  def self.config= config
+    @config = Configuration.new config
+  end
+
+  def self.with_temporary_config temporary_config
+    old_config = @config
+    @config = Configuration.new config.config.merge(temporary_config)
```

Comment:
Yeah, the `config.config` this is kinda gross... maybe I will rename it to `config_hash` to be a bit more obvious, but the `@config` is the options that are passed into the configuration that override the defaults.  Since the defaults are not merged with the config that is passed in, and we just evaluate if the value exists or defer to the default when requesting the value, we are just merging what is passed into the initial config instance and updating it with the temp changes.